### PR TITLE
NK/test-log-output-cleanup

### PR DESCRIPTION
**Pull request by nikola-klobucar**
This PR was made so all tests where `AppLogger.with_log_level(:silence)` in Infoswitch/Gateway are used to see what's a better approach. In this PR, I have created a new method that takes 2 elements in an array (severity and message) and checks if both elements match the excepted `error/warning/debug` etc... If just one of them does not match the excepted stout, the logger if write the message on a terminal/console. Otherwise it will not.

This was made so if certain errors occur during the development that are not expected in a particular test, the Logger will print it out and we will be able to larger control on error handling as well as in debugging process.

### CODE REVIEW

**Comment by nikola-klobucar** [comment]
without-applogger2
with-new-funcionality
with-new-func-terminal

The first image shows test without any Applogger functionality. Second one is how the new functionality is implemented and 3rd are its results.

wrong-excepted-message
wrong-excepted-error-terminal

In 4th image, I have purposely 'excepted' a wrong error message and 5th image shows that only that test have logged an error as it never had the filtered fuctionality.

assert-simple-reversal
assert-simple-reversal-terminal

However, in some other tests, although the functionality 'suffocates' excepted messages, some other log types are still showing. A test `test_error_in_commit_block` in `authorize_or_purchase_test.rb` and `capture_test.rb` was used to showcase this. This is an indicator that functionality needs some kind of upgrade.

File: test/unit/app_logger_test.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        begin
+          raise ArgumentError.new 'Test message'
+        rescue Exception => e
+          mock_backtrace e, :warn
+        end
+      end
+    end
+
+    assert_equal true, output.include?("[warn] ArgumentError Test message\n")
+  end
```
Comment (luka2805) [inline test/unit/app_logger_test.rb:45]: I don't see a test where the one exception is suppressed and the other one(non-suppressed) is logged, which I think is the whole point of this PR

**Comment by luka2805** [comment]
Also don't forget to update gem version in each PR @nikola-klobucar

File: test/unit/app_logger_test.rb (hunk 1 above)
Comment (nikola-klobucar) [reply test/unit/app_logger_test.rb:45]: This current setup is de facto MVP. Further development will be commited in following days.

Comment (luka2805) [reply test/unit/app_logger_test.rb:45]: Test for new functionality should be present even before functionality is implemented. https://www.youtube.com/watch?v=ln4WnxX-wrw

Comment (nikola-klobucar) [reply test/unit/app_logger_test.rb:45]: I know the basic of TDD. But Thank you.

**Pull request closed without merging**